lib/temporal/failure_converter/base.rb

```diff
@@ -0,0 +1,13 @@
+module Temporal
+  module FailureConverter
+    class Base
+      def to_failure(_error)
```

Comment:
You (may) be happy to know, but after more off-PR discussion with @bergundy, we're gonna follow the original suggestion of passing in the payload converter to each call instead of the awkward way now.